traceevents/src/main/java/com/tomtom/kotlin/traceevents/TraceEvent.kt

```diff
@@ -46,19 +45,33 @@ public data class TraceEvent(
     val stackTraceHolder: Throwable?,
     val eventName: String,
     val args: Array<Any?>,
-    val parameterNames: Array<String>?
+    internal val parameterNamesProvider: () -> Array<String>?
 ) {
-    init {
-        check(parameterNames == null || args.size == parameterNames.size)
+    /**
+     * Names of the parameters, in the same order as the `args` values. Normally, you would use [getNamedParametersMap]
+     * to access the parameters names and values. If `null` the map could not be created for some reason.
+     * You can still use the `args` array in that case.
+     *
+     * Obtaining the parameters names requires reflection.
```

Comment:
Unclear what the significance of this is for a caller.

Reply:
Explained in doc.